Sources/SwiftBasicFormat/Trivia+Indented.swift

```diff
@@ -13,25 +13,33 @@
 import SwiftSyntax
 
 extension Trivia {
-  func indented(indentation: TriviaPiece) -> Trivia {
+  func indented(indentation: TriviaPiece, isOnNewline: Bool = false) -> Trivia {
     var indentedPieces: [TriviaPiece] = []
     for (index, piece) in self.enumerated() {
-      let nextPiece = index < pieces.count - 1 ? pieces[index + 1] : nil
-      indentedPieces.append(piece)
-      if piece.isNewline {
-        switch (nextPiece, indentation) {
-        case (.spaces(let nextPieceSpaces)?, .spaces(let indentationSpaces)):
+      let previousPieceIsNewline: Bool
+      if index == 0 {
+        previousPieceIsNewline = isOnNewline
+      } else {
+        previousPieceIsNewline = pieces[index - 1].isNewline
+      }
+      if previousPieceIsNewline {
+        switch (piece, indentation) {
+        case (.spaces(let nextPieceSpaces), .spaces(let indentationSpaces)):
           if nextPieceSpaces < indentationSpaces {
             indentedPieces.append(.spaces(indentationSpaces - nextPieceSpaces))
           }
-        case (.tabs(let nextPieceTabs)?, .tabs(let indentationTabs)):
+        case (.tabs(let nextPieceTabs), .tabs(let indentationTabs)):
           if nextPieceTabs < indentationTabs {
             indentedPieces.append(.tabs(indentationTabs - nextPieceTabs))
           }
         default:
           indentedPieces.append(indentation)
         }
       }
+      indentedPieces.append(piece)
```

Comment:
Actually, I’m thinking we might be able to avoid the entire subtraction logic because anywhere we subtract, we will just insert the subtracted number of spaces/tabs later on. So we should be able to change this to something like

```swift
case (.spaces(let nextPieceSpaces), .spaces(let indentationSpaces)):
   if nextPieceSpaces < indentationSpaces {
     indentedPieces.append(indentation)
   } else {
     indentedPieces.append(piece)
   }
 case (.tabs(let nextPieceTabs), .tabs(let indentationTabs)):
   if nextPieceSpaces < indentationSpaces {
     indentedPieces.append(indentation)
   } else {
     indentedPieces.append(piece)
   }
 default:
   indentedPieces.append(indentation + piece)
 }
}
```

I haven’t tried it though.